packages/block-library/src/query-loop/edit.js

```diff
@@ -80,28 +81,20 @@ export default function QueryLoopEdit( {
 				query.sticky = sticky === 'only';
 			}
 
-			// When you insert this block outside of the edit site then store
-			// does not exist therefore we check for its existence.
-			// TODO: remove this code, edit-site shouldn't be called in block-library.
-			// This creates a cycle dependency.
-			if ( inherit && select( 'core/edit-site' ) ) {
-				// This should be passed from the context exposed by edit site.
-				const { getEditedPostType, getEditedPostId } = select(
-					'core/edit-site'
-				);
-
-				if ( 'wp_template' === getEditedPostType() ) {
-					const { slug } = select( 'core' ).getEntityRecord(
-						'postType',
-						'wp_template',
-						getEditedPostId()
+			// If `inherit` is truthy search for queryContext information provided
+			// to match preview better.
+			if ( inherit ) {
```

Comment:
This is a great start. It decouples this block from the edit site store. It's still not clear how this context can be set outside of the edit site but at least it can be achieved when necessary :)

Reply:
This code is only for site editor, for preview purposes. In a single page, Query should be custom as there is no need/reason to have a Query which inherits the global context.

Reply:
>  In a single page, Query should be custom as there is no need/reason to have a Query which inherits the global context.

You still would need to find a way to disable the global context for the Query Loop block when inserted into a single post type to achieve this goal. I guess it's possible through detecting the post type, although with custom post types it might not that simple.

Reply:
You're right about this. 🤔